### DISCUSSION TITLE
How to bind or-tools to golang project

### DISCUSSION THREAD

**Discussion by komalbb** [google/or-tools#2581] [Build and configuration issues] [answered]
Hi All,

I am trying to use or-tools in my golang project. The or-tools officially hasn't supported golang bindings.
I am new to use bindings;

I tried following:
i. wget https://github.com/google/or-tools/releases/download/v9.0/or-tools_ubuntu-18.04_v9.0.9048.tar.gz 
ii. in golang project; created a folder vrp, created vrp.go and vrp.swagcxx and vrp.h and copied include/ortools/constraint_solver to vrp folder and included them In vrp.h
iii. in vrp.swgcxx  // See swig.org for more inteface options,
// e.g. map std::string to Go string

%{
#include "vpr.h"
%}

%include "vpr.h"

iv. in vrp.go waant to access the routing API. 

Is this correct way of binding the or tools to golang?
how to call functions of routing from golang code.

Your inputs are really appreciated. Thanks

**Comment by ghost** [comment, marked as answer]
Related: https://github.com/google/or-tools/issues/953
